js/main.js

```diff
@@ -1,374 +0,0 @@
-// Not using preload here, to avoid loadingscreen
-function loadBefore() {
-  font = loadFont("fonts/MPLUSROUNDED1C-BOLD.TTF");
-  // sounds
-  bgmusic = loadSound("sounds/background2.mp3");
-  minimize = loadSound("sounds/minimize.wav");
-  maximize = loadSound("sounds/maximize.wav");
-  bop = loadSound("sounds/bop.wav");
-  succes = loadSound("sounds/succes.wav");
-  gameover = loadSound("sounds/gameover.wav");
-  infobop = loadSound("sounds/infobop.wav");
-}
-
-function setup() {
-  createCanvas(750, 900);
-  loadBefore();
-  drawVolumeSlider();
-  resetSketch(true, 0);
-}
-
-function draw() {
-  outputVolume(slider.value());
-  if (startScreenShow) {
-    startScreen();
-  } else if (gameOver) {
-    gameOverScreen();
-  } else if (!startScreenShow) {
-    background(30);
-    background(colorChange, 0, 255, colorChange / 10);
-    drawBackground();
-    updatePlayer();
-    updateObstacleAndParticles();
-    gameStatus();
-    drawScore();
-    spawnNewLevel();
-    // debugging();
-  }
-}
-
-// Spwawns new level
-function spawnNewLevel() {
-  if (levelCounter === 0) spawnLevelOne();
-  if (levelCounter === 1) spawnLevelTwo();
-  if (levelCounter === 2) spawnLevelThree();
-  if (levelCounter === 3) spawnLevelFour();
-  if (levelCounter === 4) spawnLevelFive();
-  if (levelCounter === 5) spawnLevelSix();
-  if (levelCounter === 6) spawnLevelSeven();
-}
-
-// Filters and updates the RandomParticleArray
-function updateRandomParticles() {
-  randomParticles = randomParticles.filter((particle) => {
-    particle.update();
-    particle.move();
-    return !particle.collided;
-  });
-
-  if (randomParticles.length < 20) {
-    for (let i = 0; i < 1; i++) {
-      randomParticles.push(new RandomParticle());
-    }
-  }
-}
-
-// Filters and updates the RandomParticleArray
-function updateThings() {
-  thingsArray.forEach((thing) => {
```

Comment:
thingsArray is such a bad name for this. group "things" into meaningful groups and give those a name.

Reply:
i.e. if I don't know your game code, what is thingsArray? Is it the world, the particles in it? The player? What about obstacles... from the name I don't know ^^